### PR TITLE
format irregular map keys

### PR DESCRIPTION
**Pull request by woylie**
I just noticed that `JasonSafeFormatter` won't format irregular map keys. I added a test case for this, but I'm not sure what the result should be in this case. The test causes a `Protocol.UndefinedError`.

related to #73

### CODE REVIEW

File: test/unit/logger_json_datadog_test.exs [hunk 1]
```diff
@@ -235,14 +235,14 @@ defmodule LoggerJSONDatadogTest do
...
-    %{module: mod, function: {name, arity}, file: file, line: line} = __ENV__
+    %{module: mod, function: {name, arity}, file: _file, line: _line} = __ENV__
 
     log =
       fn -> Logger.debug("hello") end
       |> capture_log()
       |> Jason.decode!()
 
-    line = line + 3
+    # line = line + 3
```
Comment (AndrewDryga) [inline test/unit/logger_json_datadog_test.exs:245]:
```suggestion
```

**Comment by AndrewDryga** [comment]
@woylie thank you, another great catch! Can you please remove than one comment line and I'll merge it

**Comment by woylie** [comment]
I only added a failing test case. How would you suggest to format such a map key? Maybe inspect it, so that we have a string key?

**Comment by AndrewDryga** [comment]
@woylie my review was too fast, sorry. Almost missed it :/.

I think you are right, the only way to encode it is to use `inspect/1`.

**Comment by woylie** [comment]
I updated the PR. I added a property test with StreamData to ensure we don't miss anything (and I would have without it). This should be ready to merge now.

**Comment by woylie** [comment]
Just pushed some more minor fixes, and updated the travis config so it doesn't fail.

**Comment by woylie** [comment]
The reported coverage dropped after upgrading excoveralls to the latest version.

**Comment by woylie** [comment]
I'm done here, should be ready to merge.

**Review by AndrewDryga — APPROVED** [review]

**Comment by AndrewDryga** [comment]
Thank you @woylie ❤️

**Pull request merged**